docs/book/src/cronjob-tutorial/testdata/project/api/v1/cronjob_webhook.go

```diff
@@ -46,6 +49,8 @@ Then, we set up the webhook with the manager.
 func (r *CronJob) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
+		WithValidator(&CronJobCustomValidator{}).
```

Comment:
I think there is no need why two separate types are needed for validator/defaulter (but also no need to just have one)

I personally would put them on the same type

Reply:
has scenarios where this types will be properly filled?
I mean, have specs? What is the purpose of the type?

Reply:
Not sure I understand the question. These type are only there to hold the webhook implementation. They won't have a "spec"

Reply:
I'm fine either way, just sharing how we implemented it

Reply:
Thank you. I will give a look !!

Reply:
After think about this one further, I've reach out the idea that we need to include the resource `Kind` directly within the `CustomValidator` and `CustomDefaulter` structs by default. This is because these structs will frequently interact with the resource's fields during validation and defaulting processes. (as we are updating now the cronjob sample)

In addition to including the `Kind`, we need to maintain both structures (`CustomValidator` and `CustomDefaulter`). Each serves distinct purposes that are often necessary in practice. For example:

- **CustomDefaulter:** May contain fields that provide default values not only for the resource itself but also by referencing other resources or objects that carry default settings. Example:  We could include a `DefaultConcurrencyPolicy` that automatically applies a default value if the user doesn't specify one. Additionally, references to other resources, such as a global configuration object, could supply environment-specific defaults.
- **CustomValidator:** Often requires fields that help enforce policies or ensure that the resource meets specific criteria before it is allowed to be created, updated, or deleted. Example; Fields like `MaxConcurrentJobs` or `AllowedNamespaces` could be used to enforce cluster policies, ensuring that the resource operates within predefined limits or is only deployed in approved environments.

Reply:
Done !